Project/Src/Gui/TextAreaDragDropHandler.cs

```diff
@@ -93,24 +96,35 @@ protected void OnDragDrop(object sender, DragEventArgs e)
                     var offset = textArea.Caret.Offset;
                     if (textArea.IsReadOnly(offset))
                         return;
-                    if (e.Data.GetDataPresent(typeof(DefaultSelection)))
+
+                    try
                     {
-                        var sel = (ISelection)e.Data.GetData(typeof(DefaultSelection));
-                        if (sel.ContainsPosition(textArea.Caret.Position))
-                            return;
-                        if (GetDragDropEffect(e) == DragDropEffects.Move)
+                        if (e.Data.GetDataPresent(typeof(DefaultSelection)))
                         {
-                            if (SelectionManager.SelectionIsReadOnly(textArea.Document, sel))
+                            var sel = (ISelection)e.Data.GetData("ICSharpCode.TextEditor.Document.DefaultSelection");
+                            if (sel.ContainsPosition(textArea.Caret.Position))
                                 return;
-                            var len = sel.Length;
-                            textArea.Document.Remove(sel.Offset, len);
-                            if (sel.Offset < offset)
-                                offset -= len;
+                            if (GetDragDropEffect(e) == DragDropEffects.Move)
+                            {
+                                if (SelectionManager.SelectionIsReadOnly(textArea.Document, sel))
+                                    return;
+                                var len = sel.Length;
+                                textArea.Document.Remove(sel.Offset, len);
+                                if (sel.Offset < offset)
+                                    offset -= len;
+                            }
                         }
                     }
-
+                    catch (System.InvalidCastException)
+                    {
+                        // If GetDataPresent(typeof(DefaultSelection)) threw this exception, then it's an interprocess DefaultSelection COM object that is not serializable!
```

Comment:
Text should be wrapped much closer to 80 chars, it is hard to read on GitHub

Reply:
Okay, I will make the fix